app/models/user.rb

```diff
@@ -0,0 +1,15 @@
+class User < ApplicationRecord
+  ROLES = { sale: 0, admin: 1, moderator: 2 }.freeze
+  enum role: ROLES
+  before_save { email.downcase! }
+  validates :first_name, length: { minimum: 3, maximum: 50 }, presence: true
+  validates :last_name,  length: { minimum: 3, maximum: 50 }, presence: true
+  validates :password,   length: { minimum: 6, maximum: 25 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, length: { maximum: 255 },
```

Comment:
why do you need this if you have already devise gem
please look at it's sources (validatable module) https://github.com/plataformatec/devise/blob/master/lib/devise/models/validatable.rb#L29-L36